app/controllers/admin/api/api_docs_backend_apis_controller.rb

```diff
@@ -0,0 +1,128 @@
+    respond_with(api_docs_backend_api)
+  end
+
+  # swagger
+  ##~ sapi = source2swagger.namespace("Account Management API")
+  ##~ e = sapi.apis.add
+  ##~ e.path = "/admin/api/backend_api_active_docs/{id}.json"
+  ##~ e.responseClass = "backend_api_active_doc"
+  #
+  ##~ op            = e.operations.add
+  ##~ op.httpMethod = "PUT"
+  ##~ op.summary    = "Backend API ActiveDocs Spec Update"
+  ##~ op.description = "Updates the Backend API ActiveDocs spec by ID"
+  ##~ op.group = "backend_api_active_docs"
+  #
+  ##~ op.parameters.add @parameter_access_token
+  ##~ op.parameters.add @parameter_active_backend_api_doc_id_by_id
+  ##~ op.parameters.add :name => "name", :description => "Name of the Backend API ActiveDocs spec", :required => true, :dataType => "string", :paramType => "query"
+  ##~ op.parameters.add @parameter_backend_api_id
+  ##~ op.parameters.add :name => "body", :description => "Backend API ActiveDocs specification in JSON format (based on Swagger)", :dataType => "string", :required => true, :paramType => "query"
+  ##~ op.parameters.add :name => "description", :description => "Description of the Backend API ActiveDocs spec", :dataType => "string", :paramType => "query"
+  ##~ op.parameters.add :name => "published", :description => "Set to 'true' to publish the spec on the developer portal, or 'false' to hide it", :dataType => "boolean", :paramType => "query"
+  ##~ op.parameters.add :name => "skip_swagger_validations", :description => "Set to 'true' to skip validation of the Swagger specification, or 'false' to validate the spec", :dataType => "boolean", :paramType => "query"
+  #
+  def update
+    api_docs_backend_api = api_docs.find(params[:id])
+    api_docs_backend_api.update(api_docs_params, without_protection: true)
+    respond_with(api_docs_backend_api)
+  end
+
+  # swagger
+  ##~ sapi = source2swagger.namespace("Account Management API")
+  ##~ e = sapi.apis.add
+  ##~ e.path = "/admin/api/backend_api_active_docs/{id}.json"
+  ##~ e.responseClass = "backend_api_active_doc"
+  #
+  ##~ op            = e.operations.add
+  ##~ op.httpMethod = "DELETE"
+  ##~ op.summary    = "Backend API ActiveDocs Spec Delete"
+  ##~ op.description = "Deletes the Backend API ActiveDocs spec by ID"
+  ##~ op.group = "backend_api_active_docs"
+  #
+  ##~ op.parameters.add @parameter_access_token
+  ##~ op.parameters.add @parameter_active_backend_api_doc_id_by_id
+  #
+  def destroy
+    api_docs_backend_api = api_docs.find(params[:id])
+    api_docs_backend_api.destroy
+    respond_with(api_docs_backend_api)
+  end
+
+  protected
+
+  def api_docs
+    current_account.all_api_docs.owned_by_backend_api
+  end
+
+  def api_docs_params(*extra_params)
+    permit_params = %i[name body description published skip_swagger_validations backend_api_id] + extra_params
+    permitted_params = params.require(:api_docs_backend_api).permit(*permit_params)
+
+    backend_api_id = permitted_params.delete(:backend_api_id)
+    if backend_api_id.present?
```

Comment:
I think the `backend_api_id` should not be optional here. Related to https://github.com/3scale/porta/pull/2116/files?file-filters%5B%5D=.rb#r504510783